### PR TITLE
Add remaning lifecycle hooks

### PR DESCRIPTION
**Pull request by slindberg**

### CODE REVIEW

**Comment by rwjblue** [comment]
Looks good to me.  I agree that it makes sense to hold off on the "extra" hooks until we have a valid use case for them.

File: README.md [hunk 1]
```diff
@@ -112,41 +112,118 @@ Options can be specified on the command line or in `config/release.js` unless ma
...
+
+  // For details on each option run `ember help release`
+  module.exports = {
+    // Build the project in the production environment, outputting to dist/
+    beforeCommit: function(project) {
+      var task = new BuildTask({
+        project: project,
+        ui: project.ui,
+        analytics: project.cli.analytics
+      });
```
Comment (slindberg) [inline README.md]: @rwjblue is there a better way to instantiate a task? Requiring the task directly seems ok-ish, but having to know what options to instantiate with it is pretty brittle.

**Comment by slindberg** [comment]
@lukemelia FYI, I've been holding off on creating a release because I wanted to validate that the build and publish stories are achievable using hooks.

**Comment by lukemelia** [comment]
@slindberg No problem. I'm running off master for the time being and things are working well!

File: README.md (hunk 1 above)
Comment (rwjblue) [inline README.md]: You do not need to require the task, but you do need to pass through the correct options.  I agree that a nice helper method in ember-cli would make that easier/better.

---

Example of avoiding require:

```javascript
module.exports = {
  // Build the project in the production environment, outputting to dist/
  beforeCommit: function(project) {
    var task = new this.tasks.Build({
      project: project,
      ui: project.ui,
      analytics: project.cli.analytics
    });

    return task.run({
      environment: 'production',
      outputPath: 'dist/'
    });
  }
};
```

Comment (slindberg) [inline README.md]: Currently hooks are not invoked with the command context, which I think is probably a good thing. It doesn't look like the `tasks` hash is available on the project anywhere either.

Seems like projects could use a `runTask(name, options)` method, although when it comes to ember cli it's difficult to tell what is considered private or not, i.e. tasks (or if there is really an official distinction).

Unless there's another way to get at that `tasks` hash, I'll consider this good enough for now.

Comment (rwjblue) [inline README.md]: Yeah, agreed (forgot about the context being wrong in the hooks). 

Also, I think at some point in the future, building might be common enough for a true/false flag in the command line or config file.

**Pull request merged**